docs/tools/terraform/v1v2.rst

```diff
@@ -0,0 +1,216 @@
+
+      kafka_user_config {
+        kafka_connect   = true
+        kafka_rest      = true
+        kafka_version   = "2.8"
+        schema_registry = true
+
+        kafka {
+          group_max_session_timeout_ms = 70000
+          log_retention_bytes          = 1000000000
+          auto_create_topics_enable    = true
+        }
+        public_access {
+          kafka_rest    = true
+          kafka_connect = true
+          prometheus    = true
+        }
+      }
+    }
+
+``terraform init``
+
+``terraform plan``
+
+``terraform apply``
+
+Upgrade Terraform
+-----------------
+
+Kick things off with 0.13
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you are using more providers than Aiven provider and/or your module
+syntax is 0.12, then you need to upgrade your modules and providers
+first by installing Terraform v0.13.x (i.e. 0.13.7):
+``tfenv install 0.13.7 && tfenv use 0.13.7``
+
+You will likely have an existing state file, by running:
+``terraform state replace-provider registry.terraform.io/-/aiven registry.terraform.io/aiven/aiven``
+you will replace old Aiven terraform provider references to the new
+format.
+
+Also update ``required_version`` from ``>= 0.12`` to ``>= 0.13`` in your
+``versions.tf`` file if you have any.
+
+We can remove the old Terraform folder ``rm -rf ~/.terraform.d``
+
+Edit the providers block of your script to include the latest version of
+the Aiven Terraform Provider (v2.3.1 as of 5th November 2021)
+
+::
+
+    terraform {
+      required_providers {
+        aiven = {
+          source = "aiven/aiven"
+          version = "2.3.1"
+        }
+      }
+    }
+
+After that you can run: ``terraform 0.13upgrade`` to see fixes
+recommended by Hashicorp. More information
+`here <https://www.terraform.io/upgrade-guides/0-13.html>`__.
+
+Bump it to the latest
+~~~~~~~~~~~~~~~~~~~~~
+
+Any version above 0.13 will be fine, here we use the latest (1.0.10 as
+of 5th November 2021) ``tfenv install latest && tfenv use latest``
+
+Run ``terraform init -upgrade``
+
+Run ``terraform plan``
+
+.. image:: /images/tools/terraform/terraform-upgrade.jpg
+   :alt:
+
+You may see warnings or errors like the above, these will point towards
+changes made between the release you are running and the latest release.
+
+The warnings will provide recommendations on the changes to make and you
+can get more information using our
+`docs <https://registry.terraform.io/providers/aiven/aiven/latest/docs>`__.
+
+Migrating away from ``aiven_service`` to the ``aiven_X`` service specific resource
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Migration strategy – since ``aiven_service`` and ``aiven_X``, i.e.
+``aiven_kafka`` are different kind of resources just by rewriting code
+we would cause destructive actions.
+
+Also running ``terraform state mv <a> <b>`` will not do it because it is
+a different resource type.
+
+To solve it we would have to:
+
+-  Change the code first
```

Comment:
I think this step is covered in this section - but what about the other two steps?

Reply:
Yep, it implies updating TF version to the latest available.

Reply:
Here they are – https://github.com/aiven/devportal/pull/343/files/1d46af02cbe13d06a8217b8d20de341afb1bfb1c#diff-09583ec3f3df84677dd731f7ab9ec32f58af3b7d2f2e0f85b59ae048c236f9faR201